Mathlib/Combinatorics/SimpleGraph/Acyclic.lean

```diff
@@ -152,4 +155,57 @@ theorem isTree_iff_existsUnique_path :
       simp only [ExistsUnique.unique (h v w) hp hq]
 #align simple_graph.is_tree_iff_exists_unique_path SimpleGraph.isTree_iff_existsUnique_path
 
+lemma IsTree.existsUnique_path (hG : G.IsTree) : ∀ v w, ∃! p : G.Walk v w, p.IsPath :=
+  (isTree_iff_existsUnique_path.1 hG).2
+
+lemma IsTree.card_edgeFinset [Fintype V] [Fintype G.edgeSet] (hG : G.IsTree) :
+    Finset.card G.edgeFinset + 1 = Fintype.card V := by
```

Comment:
I'm a bit sad to see `Fintype` and `Finset.card` instead of `Finite` and `Nat.card` but poking around the graph theory library I see this is consistent with the rest.

What view do @b-mehta @YaelDillies @kmill @bottine have on refactoring to the classical world?

Reply:
@ocfnash The graph theory library pre-dates `Finite` (and you might notice that I'm the author of `Finite`, so you might be able to guess my general thoughts :wink:). I hope eventually we can use `Finite`, but that'll take switching from things like `Finset.sum` to suitable `Set` counterparts.

Reply:
`finset` works great for the graph theory library. I don't think there's any switching to be done, but I agree that we should have nicer ways to say "This graph is finite" than spelling out all the fintype instaces.